gms/gossiper.cc

```diff
@@ -1431,6 +1431,10 @@ endpoint_state& gossiper::get_or_create_endpoint_state(inet_address ep) {
         if (utils::fb_utilities::is_me(ep)) {
             auto rpc_addr = utils::fb_utilities::get_broadcast_rpc_address();
             eps.add_application_state(application_state::RPC_ADDRESS, versioned_value::rpcaddress(rpc_addr));
+            auto node = get_token_metadata_ptr()->get_topology().this_node();
+            if (node && node->host_id()) {
+                eps.add_application_state(application_state::HOST_ID, versioned_value::host_id(node->host_id()));
```

Comment:
What's the reason for this change?

Reply:
prep for https://github.com/scylladb/scylladb/issues/12282

Reply:
But you don't have to add HOST_ID to create an endpoint state for a given host_id.

Reply:
Why complicate our lives?
If the endpoint_state is guaranteed to contain the HOST_ID it is self-identifyable.

Reply:
Maybe it will make more sense in the context of #15444